zerver/management/commands/deliver_scheduled_messages.py

```diff
@@ -62,12 +61,10 @@ def handle(self, *args: Any, **options: Any) -> None:
                 messages_to_deliver = ScheduledMessage.objects.filter(
                     scheduled_timestamp__lte=timezone_now(),
                     delivered=False)
-                if messages_to_deliver:
-                    for message in messages_to_deliver:
-                        with transaction.atomic():
-                            do_send_messages([self.construct_message(message)])
-                            message.delivered = True
-                    Message.objects.bulk_update(messages_to_deliver, ['delivered'])
+                for message in messages_to_deliver:
+                    do_send_messages([self.construct_message(message)])
+                    message.delivered = True
+                    message.save(update_fields=['delivered'])
```

Comment:
Should we keep taking advantage of `bulk_update`, e.g. by storing the successfully `delivered` messages  in a list and doing `bulk_update` in a `finally` block?

Reply:
I think not, since the previous model (A) doesn't materially improve performance and (B) is it risk of resulting in us resending several messages in the event of an exception.